src/plugins/tool_extrude.py

```diff
@@ -54,11 +56,29 @@ def on_drag_start(self, target):
                 col = target.voxels.get(x, y, z)
                 self._stamp.append((x, y, z, col))
         self._mouse = (target.mouse_x, target.mouse_y)
-        self.xdir = True
-        self.ydir = True
-        self.zdir = True
+        if QtCore.Qt.Key_X in target.keys:
+            self.xdir = True
+            self.ydir = False
+            self.zdir = False
+            self.fixeddirection = True
+        elif QtCore.Qt.Key_Y in target.keys:
+            self.xdir = False
+            self.ydir = True
+            self.zdir = False
+            self.fixeddirection = True
+        elif QtCore.Qt.Key_Z in target.keys:
+            self.xdir = False
+            self.ydir = False
+            self.zdir = True
+            self.fixeddirection = True
+        else:
+            self.xdir = True
+            self.ydir = True
+            self.zdir = True
+            self.fixeddirection = False
```

Comment:
I'd guess it's a bug in the directional lock logic and the view angle matters because the extrude direction would be a different one without holding down the direction lock key. It's still easily reproducible with the latest commit.